comps/integrations/haystack/src/opea_haystack/embedders/tei/text_embedder.py

```diff
@@ -0,0 +1,144 @@
+from typing import Any, Dict, List, Optional, Union
+
+from haystack import component, default_from_dict, default_to_dict
```

Comment:
Just as in the document embedder, we don't need `default_from_dict` here
```suggestion
from haystack import component, default_to_dict
```